agent/bench-scripts/pbench-uperf

```diff
@@ -181,280 +182,295 @@ function gen_xml {
 	echo "</profile>"
 }
 
+function is_remote {
+	local arg="${1}"
+	local _remote
+	if [[ "${arg}" == "127.0.0.1" || "${arg}" == "localhost" ]]; then
+		# False, non-zero return code
+		_remote=1
+	else
+		# True, zero return code
+		_remote=0
+	fi
+	return ${_remote}
+}
+
 function stop_server {
-	local server=$1
-	local server_port=$2
-	local message=$3
-	local uperf_pids=`ssh $ssh_opts $server ss -tlnp --vsock | sed -En -e "/:$server_port / s/.*pid=([^,]+),.*/\1/p" | sort -u`
-	if [ ! -z "$uperf_pids" ]; then
-		if [ $message -eq 1 ]; then
-			echo "found uperf pid(s) $uperf_pids on $server:$server_port, killing"
+	local server="${1}"
+	local server_port="${2}"
+	local message="${3}"
+
+	local ss_output
+	if is_remote ${server}; then
+		ss_output=$(ssh ${ssh_opts} ${server} ss -tlnp --vsock)
+	else
+		ss_output=$(ss -tlnp --vsock)
+	fi
```

Comment:
I *almost* hate to even suggest this tweak, but ... yeah, I'm going to anyway.

It seems that most of your `is_remote` usage is for exactly this purpose: execute a command, either locally or remotely via `ssh`. So ... *that* could be encapsulated into a function, too, e.g., `execute`, taking a command line string and either executing it directly or via `ssh`.

The result would be much more streamlined and easier to read.

Reply:
Peter [already suggested basically that](https://github.com/distributed-system-analysis/pbench/pull/2533#discussion_r745576765).  😁

Reply:
But...the encapsulation could and should use the `is_remote` function!

Reply:
I like the sentiment of this further encapsulation, but it becomes more difficult in `bash` to ensure the command makes its way through the layers of encapsulation to land on the remote host correctly in the general case.

Solving this in Python3 is much cleaner but out of the scope of this PR or of this code base as it stands in general.